docs/release_guidelines.md

```diff
@@ -1,23 +1,115 @@
+
+For example, the branch release-4.8 will be able to deploy OCP/ODF 4.8 as well as 4.7 to support upgrade testing. However, it will not officially support deployment or testing of versions <= 4.6.
+
+
+## Usage
+
+### Locally
+
+As these are simply git branches we are creating, you just need to perform a git checkout on the specific release branch you wish to run your deployment or tests with in order to use it.
+
+```
+git fetch upstream
+git checkout release-4.12
+```
+
+Note that `upstream` is the name of the remote for the base repository, not your personal fork.
+
+### Jenkins
+
+From Jenkins we will automatically be using the release branch if no other ocs-ci branch has been specified. Which specific release branch we use will be dynamically based on the `OCS_VERSION` or `UPGRADE_OCS_VERSION` parameters. For example, if `OCS_VERSION` is 4.13 we will use `release-4.13`. If this release branch doesn't exist yet, we will default back to the `stable` branch. This logic can be overridden by specifying a branch to use.
+
+## Creating Release Branches
+
+Release branches will branch off of the `stable` branch. Before we can create a release branch, we will need to know which commit to branch off from. Once we know the commit we are basing the branch off of, we can then go about creating the branch from either the git CLI or the Github UI.
+
+### Github UI
+
+From the Github UI, navigate to the list of [branches](https://github.com/red-hat-storage/ocs-ci/branches). From here you can click on new branch, selecting the `stable` branch as the base. Note that the branch creation process only works this way if the latest stable commit is the point we wish to branch off from. Otherwise we will need to create the branch using the CLI.
+
+### CLI
+
+First, ensure your local repository is up to date with the remote. Note that the following command is destructive so make sure you have any local changes stashed or committed before continuing.
+```
+git checkout master && git fetch --all && git reset --hard upstream/master
+```
+
+Checkout the stable branch
+```
+git checkout stable
+```
+
+Determine the starting point of the new release branch, for this you will need the git hash. You can find this using:
+```
+git log
+```
+
+Once you know the hash, we will create the new release branch using the hash from earlier as the base. Note that x.y is the major.minor version.
+```
+git checkout -b release-x.y 46568b9b4b8c2ceabad665b839655e0b19ab3634
+```
+
+Finally, push the branch to the remote.
+```
+git push upstream release-x.y
+```
+
+
+## Backporting Changes
+
+Most changes to the repository will end up being merged to master, propagage to the stable branch and evnetually end up on the next release branch. Critical bug fixes or important test cases may be selected to be backported to existing release branches. In order to backport changes to an existing release branch, we will need to take the following steps.
+
```

Comment:
What about new fixes related only to specific release? Maybe we should document those as well that those can be directly created on top of particular release branch and we can create PR only on specific release branch instead of fix going through master and all branches till e.g. release-4.8.  And the following rule of support `N-1` , the change should be cherrypicked to `N+1` as well. So in this case I am thinking better to create change to directly `N+1` and have it cherrypicked to `N-1` .  WDYT?

Reply:
I think if a change is necessary for the latest code/release and older code it should always be merged to master then cherry picked backwards. I don't think we should be first merging changes to previous releases then cherry picking forward to more recent releases.

In the event that we are adding a bugfix ONLY for a previous release, one that doesn't need to be merged to master, we would just raise a pull request against that release branch without the need to cherry-pick anything. I can add some doc around that, but I think this will be an uncommon case.

Reply:
As we discussed in meeting:
Commit for version 4.8 which is deployment or pre upgrade test related.
For scenario:
Upgrade from 4.8 -> 4.9
We run from:
release-4.9
The change should go first to 4.9 and cherry-picked to 4.8 release branch.

Reply:
Added a section about this edge-case below